source/base/quadrature_lib.cc

```diff
@@ -144,6 +144,234 @@ namespace internal
+    //
+    std::vector<double>
+    get_left_quadrature_points(const unsigned int n)
+    {
+      std::vector<double> q_points(n);
+      switch (n)
+        {
+          case 1:
+            q_points[0] = 0.;
+            break;
+          case 2:
+            q_points[0] = 0.;
+            q_points[1] = 2. / 3.;
+            break;
+          case 3:
+            q_points[0] = 0.;
+            q_points[1] = (6. - std::sqrt(6)) * 0.1;
+            q_points[2] = (6. + std::sqrt(6)) * 0.1;
+            break;
+
+          case 4:
+            q_points[0] = 0.000000000000000000;
+            q_points[1] = 0.212340538239152943;
+            q_points[2] = 0.590533135559265343;
+            q_points[3] = 0.911412040487296071;
+            break;
+          case 5:
+            q_points[0] = 0.000000000000000000;
+            q_points[1] = 0.139759864343780571;
+            q_points[2] = 0.416409567631083166;
+            q_points[3] = 0.723156986361876197;
+            q_points[4] = 0.942895803885482331;
+            break;
+          case 6:
+            q_points[0] = 0.000000000000000000;
+            q_points[1] = 0.098535085798826416;
+            q_points[2] = 0.304535726646363913;
+            q_points[3] = 0.562025189752613841;
+            q_points[4] = 0.801986582126391845;
+            q_points[5] = 0.960190142948531222;
+            break;
+          case 7:
+            q_points[0] = 0.000000000000000000;
+            q_points[1] = 0.073054328680258851;
+            q_points[2] = 0.230766137969945495;
+            q_points[3] = 0.441328481228449865;
+            q_points[4] = 0.663015309718845702;
+            q_points[5] = 0.851921400331515644;
+            q_points[6] = 0.970683572840215114;
+            break;
+          case 8:
+            q_points[0] = 0.000000000000000000;
+            q_points[1] = 0.056262560536922135;
+            q_points[2] = 0.180240691736892389;
+            q_points[3] = 0.352624717113169672;
+            q_points[4] = 0.547153626330555420;
+            q_points[5] = 0.734210177215410598;
+            q_points[6] = 0.885320946839095790;
+            q_points[7] = 0.977520613561287499;
+            break;
+          default:
+            Assert(false, ExcNotImplemented());
+            break;
+        }
+      return q_points;
+    }
+
+    std::vector<double>
+    get_quadrature_points(const unsigned int                       n,
+                          const ::dealii::QGaussRadau<1>::EndPoint end_point)
+    {
+      std::vector<double> points(n);
+      std::vector<double> left_points = get_left_quadrature_points(n);
+      switch (end_point)
+        {
+          case ::dealii::QGaussRadau<1>::EndPoint::left:
+            return left_points;
+          case ::dealii::QGaussRadau<1>::EndPoint::right:
+            for (unsigned int i = 0; i < n; ++i)
+              {
+                points[n - i - 1] = 1. - left_points[i];
+              }
+            return points;
```

Comment:
Can you please move the declaration of `points` in here, i.e.,
```suggestion
          case ::dealii::QGaussRadau<1>::EndPoint::right:
            {
              std::vector<double> points(n);
              for (unsigned int i = 0; i < n; ++i)
                {
                  points[n - i - 1] = 1. - left_points[i];
                }
              return points;
            }
```
We aim to have the declaration near the place of use. As a side product, you avoid setting up the array when it's not needed.